src/iter/mod.rs

```diff
@@ -1183,6 +1185,25 @@ pub trait ParallelIterator: Sized + Send {
         intersperse::new(self, element)
     }
 
+    /// Mutates each item of this iterator before yielding it.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rayon::prelude::*;
+    ///
+    /// let mut par_iter = (0..5).into_par_iter().update(|mut x| {*x *= 2;});
+    ///
+    /// let doubles: Vec<_> = par_iter.collect();
+    ///
+    /// assert_eq!(&doubles[..], &[0, 2, 4, 6, 8]);
+    /// ```
+    fn update<F>(self, update_op: F) -> Update<Self, F>
```

Comment:
Also, just to keep related behavior grouped together, let's move this up after `inspect`.